src/scheduler/fifo.c

```diff
@@ -161,6 +161,7 @@ schedinit(void)
 
 	init_config();
 	parse_config(CONFIG_FILE);
+	update_preempt_params_from_copy();
```

Comment:
I went through the global route, and have kept them in a struct of their own as there is a possibility of other preemption parameters might move to the sched object and it will be easy to track down all of them if they are in one place within a struct.

Reply:
I don't think they should be in their own structure.  There is an existing structure that contains global sched attributes.  I don't think we should have a second just for preemption attributes.  This would lead to a mess if we had another one just for fairshare, etc.

Reply:
this struct is used for storing a copy of the values for the preemption parameters that are shifted to the server and will be used to set the values when they get wiped off in init_config(). 
If by the global struct that is used to store the global scheduler parameters you are referring to cstat, this struct too gets wiped off in init_config.
we can change the name of the struct from preempt_params to something like shifted_params and keep adding the new attributes to this struct as and when they are shifted to the server.

Reply:
Hmm, I'm sorry about this.  I thought there was a structure in the scheduler that stored variables like the sched_priv dir.  It turns out they are globals by themselves.  IMHO you should always program for the state of the code.  Calling it shifted_params talks about the past.  We should come up where these should live and call it something based on that.  Is cstat the wrong place?  We'd just need to re-read them after a HUP, but we have all that code since we need to read them after the scheduler starts.